interbotix_common_toolbox/interbotix_gravity_compensation/src/interbotix_gravity_compensation.cpp

```diff
@@ -0,0 +1,524 @@
+          // Disable if the joint does not support current control
+          torque_enable("single", joint_names_[i], false);
+        } else {
+          // Enable if the joint supports current control
+          torque_enable("single", joint_names_[i], true);
+        }
+      } else {
+        // Gripper joint
+        torque_enable("single", joint_names_[i], false);
+      }
+    }
+
+    // Set true the gravity compensation flag
+    enable_mutex_.lock();
+    gravity_compensation_enabled_ = true;
+    enable_mutex_.unlock();
+  } else {
+    // Enable the torque for all joints
+    for (size_t i = 0; i < joint_names_.size(); i++) {
+      torque_enable("single", joint_names_[i], true);
+    }
+  }
+
+  // Set the gravity compensation flag and return the response
+  response->success = true;
+  response->message = std::string("Gravity Compensation ");
+  response->message += request->data ? "Enabled" : "Disabled";
+  RCLCPP_INFO(this->get_logger(), response->message.c_str());
+}
+
+bool InterbotixGravityCompensation::load_motor_specs(const std::string & motor_specs)
+{
+  // Load the torque constants and current units from the motor_specs file
+  YAML::Node motor_specs_node;
+  try {
+    motor_specs_node = YAML::LoadFile(motor_specs);
+  } catch (YAML::Exception & e) {
+    RCLCPP_ERROR(
+      this->get_logger(),
+      "Failed to load the motor specs file '%s'! Error: %s",
+      motor_specs.c_str(), e.what());
+    return false;
+  }
+
+  // Load the torque constants, current units, and no load currents
+  float all = motor_specs_node["motor_assist"]["all"].as<float>();
+  float single = 0.0;
+  for (const auto & joint_name : joint_names_) {
+    if (motor_specs_node["motor_assist"][joint_name].IsDefined()) {
+      // Load the motor specs
+      torque_constants_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["torque_constant"].as<float>()
+      );
+      current_units_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["current_unit"].as<float>()
+      );
+      no_load_currents_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["no_load_current"].as<float>()
+      );
+      kinetic_friction_coefficients_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["kinetic_friction_coefficient"].as<float>()
+      );
+      static_friction_coefficients_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["static_friction_coefficient"].as<float>()
+      );
+      dither_speeds_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["dither_speed"].as<float>()
+      );
+
+      // Enable/disable the dither
+      if (!motor_specs_node["dither"].as<bool>()) {
+        kinetic_friction_coefficients_.back() = static_friction_coefficients_.back();
+        static_friction_coefficients_.back() = 0;
+        dither_speeds_.back() = 0;
+      }
+
+      // Scale the no load current according to the motor assist setting
+      if (all == -1) {
```

Comment:
Compare against value of same type
```suggestion
      if (all == -1.0) {
```